src/Detail/Importing/Source/DBALSource.php

```diff
@@ -2,28 +2,105 @@
 
 namespace Detail\Importing\Source;
 
-use Doctrine\DBAL;
+use Doctrine\DBAL\Driver as DBALDriver;
 
 abstract class DBALSource extends BaseSource
 {
     /**
-     * @var DBAL\Connection $connection
+     * @var DBALDriver\Connection $connection
      */
     protected $connection;
 
     /**
-     * @param DBAL\Connection $connection
+     * @param DBALDriver\Connection $connection
      */
-    public function __construct(DBAL\Connection $connection)
+    public function __construct(DBALDriver\Connection $connection)
     {
         $this->connection = $connection;
     }
 
     /**
-     * @return DBAL\Connection
+     * @return array
+     */
+    public function fetchAll()
+    {
+        return $this->findBy();
+    }
+
+    /**
+     * @return DBALDriver\Connection
      */
     protected function getConnection()
     {
         return $this->connection;
     }
+
+    /**
+     * @return string
+     */
+    abstract protected function getSelectString();
```

Comment:
This will not be needed here, will be moved to the database specific Adapter class (see comment above)